### PR TITLE
Recall vote fixes

### PR DESCRIPTION
**Pull request by AngryKnees**
## What is changing?
Fixes a few bugs including the transfer voted shuttle being double announced, and the timer being reset in certain edge-cases. Also, a refactor to make the code a bit more organized.

## Why these changes?
Bug bad, me squash.

### CODE REVIEW

**Comment by AngryKnees** [comment]
Forgot to mention it in the description, but I wholesale removed the parts of the transfer vote that'd set the shuttle to 10 minutes when the shuttle's time left was above 10 minutes. Or at least the parts of the code that would do that in theory, a bug actually made it so it'd only apply that if the shuttle's time left was above *100* minutes.

File: russstation/code/controllers/subsystem/vote.dm [hunk 1]
```diff
@@ -1,19 +1,30 @@
...
-		message_admins("The emergency shuttle has been force-called due to a successful crew transfer vote.")
+/datum/controller/subsystem/vote/proc/attempt_transfer_vote()
+	if(SSshuttle.emergency.mode == SHUTTLE_IDLE || SSshuttle.emergency.mode == SHUTTLE_RECALL)
+		initiate_vote("crew transfer","the server")
+	else
+		// recheck to see if the shuttle is no longer busy after ten minutes
+		set_transfer_timer(10 MINUTES)
+
+/datum/controller/subsystem/vote/proc/shuttlecall()
+	SSshuttle.emergency_no_recall = TRUE
```
Comment (unv-annihilator) [inline russstation/code/controllers/subsystem/vote.dm:24]: Should this check be set down below only upon successful shuttle recall? Currently it looks to be set even on failed votes.

Comment (AngryKnees) [reply russstation/code/controllers/subsystem/vote.dm:24]: Oop, you're correct, totally forgot to move this while I was shifting things around. That being said this does present a difficulty. the `request()` call won't call the shuttle if it's in any state but idle, called, or recalled. We'd only want to block recall if it actually calls the shuttle, but `request()` doesn't return any value to indicate its success or not.

Comment (AngryKnees) [reply russstation/code/controllers/subsystem/vote.dm:24]: Reconsidering that last comment, even if the call doesn't go through, that means the shuttle is current in some form of transit and therefore can be assumed as safe to block recall.

Comment (AngryKnees) [reply russstation/code/controllers/subsystem/vote.dm]: There were a few states that didn't fit into the above category

**Pull request merged**